src/replication/replicator.rst

```diff
@@ -154,98 +203,253 @@ the following order:
+
+`Green` vs `red` partitions states into "healthy" and "unhealthy",
+respectively. Unhealthy states indicate something has gone wrong and
+it might need user's attention.
+
+`Rectangle` vs `oval` separates "terminal" states from "non-terminal"
+ones. Terminal states are those which will not transition to other
+states any more. Informally, jobs in a terminal state will not be
+retried and don't consume memory or CPU resources.
+
+ * ``Initializing``: Indicates replicator has noticed the change
+   from the replication document. Jobs should transition quickly
+   through this state. Being stuck here for a while could mean there
+   is an internal error.
+
+ * ``Failed``: Replication document could not be processed and turned
+   into a valid replication job for the scheduler. This state is
+   terminal and requires user intervention to fix the problem. Typical
+   reasons for ending up in this state is a malformed document. For
+   example specifying an integer for a parameter which accepts a
+   boolean. Another reason could be specifying a duplicate
+   replication. A duplicate replication is a replication with
+   identical parameters but a different document ID.
+
+ * ``Error``: Replication document update could not be turned into a
+   replication job. Unlike the ``Failed`` state, this one is
+   temporary, and replicator will keep retrying periodically. There is
+   an exponential backoff applied in case of consecutive failures.
+   The main reason this state exists is to handle filtered
+   replications with custom user functions. Filter function content is
+   needed in order to calculate the replication ID. A replication job
+   could not be created until the function code is retrieved. Because
+   retrieval happens over the network, temporary failures have to be
+   handled.
+
+ * ``Running``: Replication job is running normally. This means,
+   there might be a change feed open, and if changes are noticed, they
+   would be processed and posted to the target. Job is still
+   considered ``Running`` even if its workers are currently not
+   streaming changes from source to target and are just waiting on the
+   change feed.
+
+ * ``Pending``: Replication job is not running and is waiting its
+   turn. This state is reached when the number of replication jobs
+   added to the scheduler exceeds ``replicator.max_jobs``. In that
+   case scheduler will periodically stop and start subsets of jobs
+   trying to give each one a fair chance at making progress.
+
+ * ``Crashing``: Replication job has been successfully added to the
+   replication scheduler. However an error was encountered during the
+   last run. Error could be a network failure, a missing source
+   database, a permissions error, etc. Repeated consecutive crashes
+   result in an exponential backoff. This state is considered temporary
+   (non-terminal) and replication jobs will be periodically
+   retried. Maximum backoff interval is around a day or so.
```

Comment:
Not as easily. We don't explicitly schedule a job to run in the future, say something like "this job shall run again at 5pm later today" instead every time we reschedule we look in the history of each job and see how many consecutive crashes it has experienced.